web/src/test/java/com/microsoft/applicationinsights/web/extensibility/modules/WebRequestTrackingTelemetryModuleTests.java

```diff
@@ -140,6 +157,326 @@ public void testUserAgentIsBeingSet() throws Exception {
         Assert.assertEquals(HttpHelper.TEST_USER_AGENT, requestTelemetry.getContext().getUser().getUserAgent());
     }
 
+    @Test
+    public void testCrossComponentCorrelationHeadersAreCaptured() {
+        
+        //setup: initialize a request telemetry context
+        RequestTelemetryContext context = new RequestTelemetryContext(DateTimeUtils.getDateTimeNow().getTime());
+        ThreadContext.setRequestTelemetryContext(context);
+
+        //mock a servlet request with cross-component correlation headers
+        Hashtable<String, String> headers = new Hashtable<String, String>();
```

Comment:
Why HashTable instead of HashMap ?

Reply:
why not? :) it's a test, let the old dog shine a bit.

Reply:
kidding :) changed it to hashmap.